components/match2/wikis/clashroyale/opponent_display_custom.lua

```diff
@@ -0,0 +1,140 @@
+
+		if opponent.type == Opponent.team then
+			self:createTeam(opponent.template or 'tbd', options)
+		elseif opponent.type == Opponent.solo then
+			self:createPlayer(opponent.players[1])
+		elseif opponent.type == Opponent.duo then
+			self:createDuo(opponent)
+		elseif opponent.type == Opponent.literal then
+			self:createLiteral(opponent.name or '')
+		end
+
+		self.root = mw.html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+CustomOpponentDisplay.BracketOpponentEntry.createTeam = OpponentDisplay.BracketOpponentEntry.createTeam
+CustomOpponentDisplay.BracketOpponentEntry.createLiteral = OpponentDisplay.BracketOpponentEntry.createLiteral
+CustomOpponentDisplay.BracketOpponentEntry.addScores = OpponentDisplay.BracketOpponentEntry.addScores
+
+function CustomOpponentDisplay.BracketOpponentEntry:createDuo(opponent)
+	local playerNode = CustomOpponentDisplay.PlayerBlockOpponent({
+		opponent = opponent,
+		overflow = 'ellipsis',
+		showLink = false,
+	})
+	self.content:node(playerNode)
+end
+
+function CustomOpponentDisplay.BracketOpponentEntry:createPlayer(player)
+	local playerNode = PlayerDisplay.BlockPlayer({
+		player = player,
+		overflow = 'ellipsis',
+		showLink = false,
+	})
+	self.content:node(playerNode)
+end
+
+function CustomOpponentDisplay.BlockOpponent(props)
+	DisplayUtil.assertPropTypes(props, CustomOpponentDisplay.propTypes.BlockOpponent, {maxDepth = 2})
+	local opponent = props.opponent
+
+	-- Default TBDs to not show links
+	props.showLink = Logic.nilOr(props.showLink, not Opponent.isTbd(opponent))
+
+	if opponent.type == Opponent.duo then
+		return CustomOpponentDisplay.PlayerBlockOpponent(props)
+	end
+
+	return OpponentDisplay.BlockOpponent(props)
+end
+
+function CustomOpponentDisplay.PlayerBlockOpponent(props)
+	local opponent = props.opponent
+
+	local playerNodes = Array.map(opponent.players, function(player)
+		return PlayerDisplay.BlockPlayer({
+			flip = props.flip,
+			overflow = props.overflow,
+			player = player,
+			showFlag = props.showFlag,
+			showLink = props.showLink,
+			showPlayerTeam = props.showPlayerTeam,
+			team = player.team,
+			abbreviateTbd = props.abbreviateTbd,
+		})
+	end)
+
+	if #opponent.players == Opponent.partySize(Opponent.solo) then
+		return playerNodes[1]
+	else
+		local playersNode = mw.html.create('div')
+			:addClass(props.showPlayerTeam and 'player-has-team' or nil)
+		for _, playerNode in ipairs(playerNodes) do
+			playersNode:node(playerNode)
+		end
+		return playersNode
+	end
+end
+
+function CustomOpponentDisplay.InlineOpponent(props)
+	DisplayUtil.assertPropTypes(props, CustomOpponentDisplay.propTypes.InlineOpponent, {maxDepth = 2})
+	local opponent = props.opponent
+
+	if opponent.type == Opponent.duo then
+		return CustomOpponentDisplay.PlayerInlineOpponent(props)
+	end
+
+	return OpponentDisplay.InlineOpponent(props)
+end
+
+function CustomOpponentDisplay.PlayerInlineOpponent(props)
```

Comment:
```suggestion
function CustomOpponentDisplay.InlinePlayer(props)
```
To follow namestandard from parent
https://github.com/Liquipedia/Lua-Modules/blob/a5919d8a26f4bf9565365f5b309b87a3b1b5ab48/components/opponent/commons/opponent_display.lua#L160-L175

Reply:
shouldn't it be plural since it is mapping players and not only 1 player
i.e. `InlinePlayers`

Reply:
Sure